### ISSUE TITLE
`data_groups` management

### ISSUE THREAD

**Issue by tnakazato** [casangi/xradio#324]
I really like the concept of `data_groups`. I have a few considerations on it. I would appreciate it if you could consider them in the future development.

* Maybe it's better to define reserved group name for common usecases such as "after applycal" or "ready for imaging"
* Any mechanism to manage chronological evolution of `data_groups` might be necessary
* There will be many versions of `data_groups` especially for data flagging. Now you have only group name to describe the group. But, as number of groups grows, users might need more detailed description on the data group. For example, they might need a field to store short description of the data group, just like [the comment field in the `flagmanager` task](https://casadocs.readthedocs.io/en/stable/api/tt/casatasks.flagging.flagmanager.html). Another option would be to make group name self-descriptive, but it would make the name too long.

**Comment by Jan-Willem** [comment]
- We can suggest some standard names. However, I think it will be hard to come up with a comprehensive list now. As we develop our software we can define that list. This would not be a breaking change to XRADIO.
- We can add date key.
- We can add a description key.

**Comment by FedeMPouzols** [comment]
Hi @tnakazato , @Jan-Willem : the branch of this issue adds a `description` and a `date` field to the `data_groups` . The MSv2 converter populates them. This also adds the data_groups attribute formally to the schema. For now the schema only gives a non-optional group: `base`, which will be required after #350 I believe.
I opened this PR:  (this is for review purposes, it could be merged there or it could be merged into other ongoing work/branches).
